### PR TITLE
add CI java http server tutorial

### PR DESCRIPTION
**Pull request by jimsheldon**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Comment by ravilach** [comment]
Thanks for the PR! @jimsheldon mind adding those tiles to the 'AllTutorials.tsx' and 'ContinuousIntegration.tsx' like you have before [since you are pro! If not I can add]. 

@KimberlyFields adding you as a CI reviewer, feel free to re-assign if needed. 

Cheers!

-Ravi

**Review by ravilach — CHANGES_REQUESTED** [review]
Tiles if possible. Cheers!

**Comment by jimsheldon** [comment]
No problem. What do you mean by "Tiles if possible"? Thanks

**Review by ravilach — APPROVED** [review]

File: tutorials/build-code/ci-java-http-server.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+   :::info
+
+   `<+pipeline.sequenceId>` is a value provided at pipeline execution time. This value matches the pipeline sequence number, which is unique to each pipeline execution.
+
+   This ensures that the resulting Docker image will always have a unique tag.
+
+   :::
+5. Click **Apply Changes**.
+
+### Add Integration Tests Stage
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:92]:
```suggestion
### Add an integration tests stage
```

File: tutorials/build-code/ci-java-http-server.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+This separate pipeline stage will pull the Docker image published in the previous step, run it as a [Background step](../../docs/continuous-integration/ci-technical-reference/background-step-settings.md), and verify the container started successfully.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:94]:
```suggestion
This separate pipeline stage pulls the Docker image that was published in the previous step, runs it as a [Background step](../../docs/continuous-integration/ci-technical-reference/background-step-settings.md), and verifies that the container started successfully.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+1. In the **Pipeline Studio** click **Add Stage** and select **Build**.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:96]:
```suggestion
1. In the **Pipeline Studio**, select **Add Stage**, and then select **Build**.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+
+   Enter `Run Connectivity Test` in the **Stage Name** field, then click **Set Up Stage**.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:98]:
```suggestion
 2. In the **Stage Name** field, enter `Run Connectivity Test`, and then select **Set Up Stage**.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+   :::tip performance tip
+
+   The steps in this stage do not require code from the Git repository. To save time in each pipeline execution, disable **Clone Codebase**.
+
+   :::
+
+2. In the **Infrastructure** tab, select **Propagate from an existing stage**, select the previous stage from the drop-down menu and click **Continue**.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:106]:
```suggestion
3. On the **Infrastructure** tab, select **Propagate from an existing stage**, select the previous stage from the drop-down menu, and then select **Continue**.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+
+### Add Background Step
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:108]:
```suggestion
### Add a background step
```

File: tutorials/build-code/ci-java-http-server.md [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+
+This Background step runs the Docker image published in the previous stage.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:110]:
```suggestion
This background step runs the Docker image that was published in the previous stage.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+
+1. Click **Add Step**, the **Step Library** dialogue window will appear. Select **Background** from the list.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:112]:
```suggestion
1. Select **Add Step**. In the **Step Library** dialog, select **Background** from the list.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+2. Enter `Run Java HTTP Server` in the **Name** field.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:113]:
```suggestion
2. In the **Name** field, enter `Run Java HTTP Server`.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+3. Expand the **Additional Configuration** section, click the **Container Registry** field and select your Docker Hub connector.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:114]:
```suggestion
3. Expand the **Additional Configuration** section, select the **Container Registry** field, and then select your Docker Hub connector.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+4. Enter `your_user/jhttp:<+pipeline.sequenceId>` (replace `your_user` with your Docker Hub username) in the **Image** field.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:115]:
```suggestion
4. In the **Image** field, etner `your_user/jhttp:<+pipeline.sequenceId>`. Replace `your_user` with your Docker Hub username.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+5. Click **+ Add** under **Port Bindings**, enter `8888` in the **Host Port** and **Container Port** fields.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:116]:
```suggestion
5. Under **Port Bindings**, select **+ Add**, and then enter `8888` in the **Host Port** and **Container Port** fields.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 13]
```diff
@@ -0,0 +1,259 @@
...
+
+   :::info
+
+   This will expose port `8888` from the running container to the host operating system. This will allow the connection test step to reach the application at `localhost:8888`.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:120]:
```suggestion
   This exposes port `8888` from the running container to the host operating system. This allows the connection test step to reach the application at `localhost:8888`.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 14]
```diff
@@ -0,0 +1,259 @@
...
+
+   :::
+
+6. Click **Apply Changes**.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:124]:
```suggestion
6. Select **Apply Changes**.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 15]
```diff
@@ -0,0 +1,259 @@
...
+
+### Add Connection Test Step
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:126]:
```suggestion
### Add a connection test step
```

File: tutorials/build-code/ci-java-http-server.md [hunk 16]
```diff
@@ -0,0 +1,259 @@
...
+
+This step runs a connection test from the host operating system, verifying the application started successfully.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:128]:
```suggestion
This step runs a connection test from the host operating system to verify that the application started successfully.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 17]
```diff
@@ -0,0 +1,259 @@
...
+
+1. Click **Add Step**, the **Step Library** dialogue window will appear. Select **Run** from the list.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:130]:
```suggestion
1. Select **Add Step**. On the **Step Library** dialog, select **Run** from the list.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 18]
```diff
@@ -0,0 +1,259 @@
...
+2. Enter `Test Connection to Java HTTP Server` in the **Name** field.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:131]:
```suggestion
2. In the **Name** field, enter `Test Connection to Java HTTP Server` .
```

File: tutorials/build-code/ci-java-http-server.md [hunk 19]
```diff
@@ -0,0 +1,259 @@
...
+3. Enter this in the **Command** field:
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:132]:
```suggestion
3. In the **Command** field, enter the following command:
```

File: tutorials/build-code/ci-java-http-server.md [hunk 20]
```diff
@@ -0,0 +1,259 @@
...
+   ```
+   until curl --max-time 1 http://localhost:8888; do
+     sleep 2;
+   done
+   ```
+   :::info
+
+   This simple connectivity test will attempt to reach the service every two seconds, until successful.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:140]:
```suggestion
   This simple connectivity test attempts to reach the service every two seconds until it is successful.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 21]
```diff
@@ -0,0 +1,259 @@
...
+
+   :::
+
+4. Click **Apply Changes**, then click **Save**.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:144]:
```suggestion
4. Select **Apply Changes**, and then select **Save**.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 22]
```diff
@@ -0,0 +1,259 @@
...
+
+   :::tip
+
+   With the application up and running, many types of tests are possible: integration, security, performance, and more!
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:148]:
```suggestion
   With the application up and running, many types of tests are possible: integration, security, performance, and more.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 23]
```diff
@@ -0,0 +1,259 @@
...
+
+   :::
+
+### Optional YAML Configuration
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:152]:
```suggestion
### Optional YAML configuration
```

File: tutorials/build-code/ci-java-http-server.md [hunk 24]
```diff
@@ -0,0 +1,259 @@
...
+      codebase:
+        connectorRef: account.Github_OAuth_XYZ123
+        repoName: your_user/jhttp
+        build: <+input>
+```
+
+</p>
+</details>
+
+## Run Your Pipeline
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:254]:
```suggestion
## Run your pipeline
```

File: tutorials/build-code/ci-java-http-server.md [hunk 25]
```diff
@@ -0,0 +1,259 @@
...
+
+1. In the **Pipeline Studio** click **Run**, the **Run Pipeline** dialogue window will open.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:256]:
```suggestion
1. In the **Pipeline Studio**, select **Run**.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 26]
```diff
@@ -0,0 +1,259 @@
...
+2. Select **Git Branch** as the **Build Type**, and enter `main` in the **Branch Name** field.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:257]:
```suggestion
2. Select **Git Branch** as the **Build Type**, and then enter `main` in the **Branch Name** field.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 27]
```diff
@@ -0,0 +1,259 @@
...
+3. Click **Run Pipeline**.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:258]:
```suggestion
3. Select **Run Pipeline**.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 28]
```diff
@@ -0,0 +1,259 @@
...
+4. Observe each step of the pipeline execution. When the first stage completes, test results will appear in the **Tests** tab. When the second stage completes, you should see the successful `curl` command in the final step.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:259]:
```suggestion
4. Observe each step of the pipeline execution. When the first stage completes, test results appear in the **Tests** tab. When the second stage completes, you should see the successful `curl` command in the final step.
```

**Review by KimberlyFields — CHANGES_REQUESTED** [review]
The content looks good, but I made several suggestions for style and formatting.

**Comment by ravilach** [comment]
@jimsheldon let us know when you are happy with your version. Looks like a few commits have been included, then can get @KimberlyFields to re-approve. Cheers!

**Review by ravilach — DISMISSED** [review]
Still good from my side for the tech changes. -Ravi

**Comment by jimsheldon** [comment]
I think all of these suggestions look good, but it's a lot of clicking to accept them...

@KimberlyFields do you have the markdown file I can just bring into my branch?

**Comment by KimberlyFields** [comment]
@jimsheldon I made the suggestions via GitHub comments. You can click **Add suggestions to batch** for each suggestion, and then commit all of the changes in one commit.

**Comment by jimsheldon** [comment]
Thanks @KimberlyFields! I didn't know that was possible.

**Review by ravilach — DISMISSED** [review]

**Pull request merged**